sklearn/neighbors/_lof.py

```diff
@@ -291,6 +291,12 @@ def fit(self, X, y=None):
             n_neighbors=self.n_neighbors_
         )
 
+        # This allows preserving input dtype for all attributes.
+        self._distances_fit_X_ = self._distances_fit_X_.astype(
+            self._fit_X.dtype,
+            copy=False,
+        )
```

Comment:
I assume that we have the same issue here, we don't want to make it lower than 32-bitness precision.

```suggestion
        if self._fit_X.dtype == np.float32:
          self._distances_fit_X_ = self._distances_fit_X_.astype(
              self._fit_X.dtype,
              copy=False,
          )
```